persist/fs/persist_manager.go

```diff
@@ -85,43 +99,58 @@ type persistManager struct {
 	sync.RWMutex
 
 	opts           Options
-	scope          tally.Scope
 	filePathPrefix string
-	rateLimitOpts  ratelimit.Options
 	nowFn          clock.NowFn
 	sleepFn        sleepFn
 	writer         FileSetWriter
-	start          time.Time
-	count          int
-	bytesWritten   int64
-	metrics        map[uint32]*shardMetrics
-	currMetrics    *shardMetrics
-
 	// segmentHolder is a two-item slice that's reused to hold pointers to the
 	// head and the tail of each segment so we don't need to allocate memory
 	// and gc it shortly after.
 	segmentHolder []checked.Bytes
+
+	status            persistManagerStatus
+	currRateLimitOpts ratelimit.Options
+
+	start        time.Time
+	count        int
+	bytesWritten int64
+	worked       time.Duration
+	slept        time.Duration
+
+	metrics persistManagerMetrics
+}
+
+type persistManagerMetrics struct {
+	writeDuration tally.Timer
```

Comment:
Yeah just feels kinda strange to use a percentile for this, why not use a gauge?